lib/src/revset.rs

```diff
@@ -318,6 +318,7 @@ pub enum RevsetCommitRef {
         remote_pattern: StringPattern,
     },
     Tags,
+    Topics(StringPattern),
```

Comment:
Perhaps, `Topics` will have to be a `RevsetFilterPredicate`, not a resolvable commit ref. Suppose topics are relevant only in mutable commits, queries like `heads(topics(..))` wouldn't need a persistent index.